Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -1180,6 +1181,99 @@ namespace AZ
             return meshHandle.IsValid() ? ToModelDataInstance(meshHandle).m_descriptor.m_customMaterials : DefaultCustomMaterialMap;
         }
 
+        AZStd::unique_ptr<StreamBufferViewsBuilderInterface> MeshFeatureProcessor::CreateStreamBufferViewsBuilder(
+            const MeshHandle& meshHandle) const
+        {
+            return AZStd::make_unique<ShaderStreamBufferViewsBuilder>(meshHandle);
+        }
+
+        DispatchDrawItemList MeshFeatureProcessor::BuildDispatchDrawItemList(
+            const MeshHandle& meshHandle,
+            const uint32_t lodIndex,
+            const uint32_t meshIndex,
+            const RHI::DrawListMask drawListTagsFilter,
+            const RHI::DrawFilterMask materialPipelineFilter,
+            DispatchArgumentsSetupCB dispatchArgumentsSetupCB) const
+        {
+            DispatchDrawItemList retList;
+
+            const AZ::RPI::MeshDrawPacketLods& drawPacketListByLod = GetDrawPackets(meshHandle);
+            const uint32_t lodCount = aznumeric_caster(drawPacketListByLod.size());
+            if (lodIndex >= lodCount)
+            {
+                // This is normal. May happen if a caller got a valid MeshHandle before
+                // a mesh is fully loaded from assets.
+                return retList;
+            }
+            const AZ::RPI::MeshDrawPacketList& drawPacketList = drawPacketListByLod[lodIndex];
+            const uint32_t meshCount = aznumeric_caster(drawPacketList.size());
+            if (meshIndex >= meshCount)
+            {
+                AZ_Error("MeshFeatureProcessor", false,
+                    "For lodIndex=%u, got invalid meshIndex=%u, maxMeshCount=%u",
+                    lodIndex, meshIndex, meshCount);
+                return retList;
+            }
+            const AZ::RPI::MeshDrawPacket& meshDrawPacket = drawPacketList[meshIndex];
+            const RHI::DrawPacket* drawPacket = meshDrawPacket.GetRHIDrawPacket();
+            const auto& shadersList = meshDrawPacket.GetActiveShaderList();
+            if (drawPacket)
+            {
+                const uint32_t drawItemCount = aznumeric_caster(drawPacket->GetDrawItemCount());
+                for (uint32_t drawItemIdx = 0; drawItemIdx < drawItemCount; ++drawItemIdx)
+                {
+                    const RHI::DrawItem* drawItem = drawPacket->GetDrawItem(drawItemIdx);
+                    if (drawItem->GetPipelineStateType() != RHI::PipelineStateType::Dispatch)
+                    {
+                        continue;
+                    }
+                    RHI::DrawListTag tag = drawPacket->GetDrawListTag(drawItemIdx);
+                    RHI::DrawFilterMask drawItemPipelineFilter = drawPacket->GetDrawFilterMask(drawItemIdx);
+                    if (
```

Comment:
Lets add a comment about this check here

Reply:
Done.